roles/provision-keycloak-apb/tasks/main.yml

```diff
@@ -1,12 +1,41 @@
 - include_tasks: label-serviceinstance.yml
 
 - include_tasks: provision-postgres.yml
-  when: POSTGRESQL_EPHEMERAL is undefined
+  when: POSTGRESQL_EPHEMERAL is undefined or
+        not POSTGRESQL_EPHEMERAL
 
 - include_tasks: provision-postgres-ephemeral.yml
-  when: POSTGRESQL_EPHEMERAL is defined
+  when: POSTGRESQL_EPHEMERAL is defined or
+        POSTGRESQL_EPHEMERAL
 
 - include_tasks: provision-keycloak.yml
+  when: SHARED_SERVICE is not defined or
+        not SHARED_SERVICE
+
+- name: "Retrieve route to keycloak"
+  shell: "oc get configmaps/{{keycloak_service_name}} -n {{ namespace }} -o jsonpath='{.data.uri}'"
+  register: keycloak_route
+  when: SHARED_SERVICE is not defined or
+        not SHARED_SERVICE
+
+- debug: var=keycloak_route
+  when: SHARED_SERVICE is not defined or
+        not SHARED_SERVICE
+
+- include_tasks: configure-keycloak.yml
+  vars:
+    KEYCLOAK_HOST="{{ keycloak_route.stdout }}"
+    KEYCLOAK_REALM="{{ namespace }}"
+  when: SHARED_HOST is not defined or
+        not SHARED_SERVICE
+
+- include_tasks: configure-keycloak.yml
+  vars:
+    KEYCLOAK_HOST="{{ SHARED_HOST }}"
```

Comment:
Do these vars get set for the configure task? is that how the above works here?

Reply:
Done!